### ISSUE TITLE
lora 9b 微调loss一直是0.0

### ISSUE THREAD

**Issue by 1148030615** [THUDM/GLM-4#279]
### System Info / 系統信息

win11 ubentu子系统

### Information / 问题信息

- [X] The official example scripts / 官方的示例脚本
- [ ] My own modified scripts / 我自己修改的脚本和任务

### Reproduction / 复现过程

5万条多伦对话数据，loss怎么一直是0.0
{'loss': 0.0, 'grad_norm': 0.0, 'learning_rate': 4.999166666666667e-05, 'epoch': 0.0}
{'loss': 0.0, 'grad_norm': 0.0, 'learning_rate': 4.998333333333334e-05, 'epoch': 0.0}
{'loss': 0.0, 'grad_norm': 0.0, 'learning_rate': 4.9975e-05, 'epoch': 0.0}
{'loss': 0.0, 'grad_norm': 0.0, 'learning_rate': 4.996666666666667e-05, 'epoch': 0.0}
{'loss': 0.0, 'grad_norm': 0.0, 'learning_rate': 4.995833333333333e-05, 'epoch': 0.0}
  0%|                                                                              | 50/60000 [00:08<1:40:46,  9.91it/s]***** Running Evaluation *****
  Num examples = 50
  Batch size = 2
  0%|                                                                              | 50/60000 [00:18<1:40:46,  9.91it/sBuilding prefix dict from the default dictionary ...█████████████████████████████████████| 25/25 [07:09<00:00, 17.89s/it]
Dumping model to file cache /tmp/jieba.cache
Loading model cost 0.284 seconds.
Prefix dict has been built successfully.
{'eval_rouge-1': 0.0, 'eval_rouge-2': 0.0, 'eval_rouge-l': 0.0, 'eval_bleu-4': 0.0, 'eval_runtime': 447.9789, 'eval_samples_per_second': 0.112, 'eval_steps_per_second': 0.056, 'epoch': 0.0}
{'loss': 0.0, 'grad_norm': 0.0, 'learning_rate': 4.995e-05, 'epoch': 0.0}
{'loss': 0.0, 'grad_norm': 0.0, 'learning_rate': 4.994166666666667e-05, 'epoch': 0.0}
{'loss': 0.0, 'grad_norm': 0.0, 'learning_rate': 4.993333333333334e-05, 'epoch': 0.0}
{'loss': 0.0, 'grad_norm': 0.0, 'learning_rate': 4.992500000000001e-05, 'epoch': 0.0}
{'loss': 0.0, 'grad_norm': 0.0, 'learning_rate': 4.991666666666667e-05, 'epoch': 0.0}
  0%|▏                                                                            | 100/60000 [07:40<1:38:31, 10.13it/s]

### Expected behavior / 期待表现

这是正常现象还是我应该停止微调训练

**Comment by zRzRzRzRzRzRzR** [comment]
max len 设置太短了或者没有用bf16训练

**Comment by 1148030615** [comment]
现在我是单卡训练，请问bf16怎么开启
data_config:
  train_file: "对话_all_glm4.jsonl"
  val_file: "对话_all_glm4_val.jsonl"
  test_file: "对话_all_glm4_val.jsonl"
  num_proc: 5
max_input_length: 8192
max_output_length: 8192
training_args:
  # see `transformers.Seq2SeqTrainingArguments`
  output_dir: ./output
  max_steps: 60000
  # needed to be fit for the dataset 5e-4
  learning_rate: 5e-4
  # settings for data loading
  per_device_train_batch_size: 2
  dataloader_num_workers: 16
  remove_unused_columns: false
  # settings for saving checkpoints
  save_strategy: steps
  save_steps: 1000
  # settings for logging
  log_level: info
  logging_strategy: steps
  logging_steps: 10
  # settings for evaluation
  per_device_eval_batch_size: 2
  evaluation_strategy: steps
  eval_steps: 50
  # settings for optimizer
  # adam_epsilon: 1e-6
  # uncomment the following line to detect nan or inf values
  # debug: underflow_overflow
  predict_with_generate: true
  # see `transformers.GenerationConfig`
  generation_config:
    max_new_tokens: 512
  # set your absolute deepspeed path here
  #deepspeed: ./configs/ds_zero_2.json
peft_config:
  peft_type: LORA
  task_type: CAUSAL_LM
  r: 8
  lora_alpha: 32
  lora_dropout: 0.1

**Comment by zRzRzRzRzRzRzR** [comment]
模型在载入的时候如果你是支持BF16的GPU就是BF16呀

**Comment by 1148030615** [comment]
我现在是要控制单条数据集长度吗，请问我需要怎么处理数据集才能正常微调

**Comment by zRzRzRzRzRzRzR** [comment]
你把配置文件的max token设置长就行

**Comment by 1148030615** [comment]
没看到配置文件里面有设置的地方， 有个max_new_tokens参数设置了 4096还是这样loss0.0

**Comment by zRzRzRzRzRzRzR** [comment]
只能自己根据源代码排查label 有没有数字了，或者子系统的问题（概率比较低）

**Comment by liuhe6** [comment]
https://github.com/THUDM/GLM-4/issues/261#issuecomment-2202254647 看看这个
